plugin/cmd/kube-scheduler/scheduler.go

```diff
@@ -36,5 +37,8 @@ func main() {
 
 	verflag.PrintAndExitIfRequested()
 
-	app.Run(s)
+	if err := app.Run(s); err != nil {
+		glog.Fatalf("scheduler app failed to run: %v", err)
+	}
+	panic("this statement shouldn't be reached")
```

Comment:
I don't think this logic makes much sense.. of course we can reach this point via a number of ways.  Shouldn't we just exit gracefully?